config/src/main/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LogoutConfigurer.java

```diff
@@ -0,0 +1,621 @@
+import org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler;
+import org.springframework.security.web.authentication.logout.SimpleUrlLogoutSuccessHandler;
+import org.springframework.security.web.csrf.CsrfFilter;
+import org.springframework.security.web.csrf.CsrfLogoutHandler;
+import org.springframework.security.web.csrf.CsrfTokenRepository;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.OrRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.util.Assert;
+
+/**
+ * Adds SAML 2.0 logout support.
+ *
+ * <h2>Security Filters</h2>
+ *
+ * The following Filters are populated
+ *
+ * <ul>
+ * <li>{@link LogoutFilter}</li>
+ * <li>{@link Saml2LogoutRequestFilter}</li>
+ * <li>{@link Saml2LogoutResponseFilter}</li>
+ * </ul>
+ *
+ * <p>
+ * The following configuration options are available:
+ *
+ * <ul>
+ * <li>{@link #logoutUrl} - The URL to initiate SAML 2.0 Logout</li>
+ * <li>{@link #logoutRequestMatcher} - The {@link RequestMatcher} to initiate SAML 2.0
+ * Logout</li>
+ * <li>{@link #logoutSuccessHandler} - The {@link LogoutSuccessHandler} to execute once
+ * SAML 2.0 Logout is complete</li>
+ * <li>{@link LogoutRequestConfigurer#logoutRequestMatcher} - The {@link RequestMatcher}
+ * to receive SAML 2.0 Logout Requests</li>
+ * <li>{@link LogoutRequestConfigurer#authenticator} - The {@link AuthenticationManager}
+ * for authenticating SAML 2.0 Logout Requests</li>
+ * <li>{@link LogoutRequestConfigurer#logoutRequestResolver} - The
+ * {@link Saml2LogoutRequestResolver} for creating SAML 2.0 Logout Requests</li>
+ * <li>{@link LogoutRequestConfigurer#logoutRequestRepository} - The
+ * {@link Saml2LogoutRequestRepository} for storing SAML 2.0 Logout Requests</li>
+ * <li>{@link LogoutResponseConfigurer#logoutRequestMatcher} - The {@link RequestMatcher}
+ * to receive SAML 2.0 Logout Responses</li>
+ * <li>{@link LogoutResponseConfigurer#authenticator} - The {@link AuthenticationManager}
+ * for authenticating SAML 2.0 Logout Responses</li>
+ * <li>{@link LogoutResponseConfigurer#logoutResponseResolver} - The
+ * {@link Saml2LogoutResponseResolver} for creating SAML 2.0 Logout Responses</li>
+ * </ul>
+ *
+ * <h2>Shared Objects Created</h2>
+ *
+ * No shared Objects are created
+ *
+ * <h2>Shared Objects Used</h2>
+ *
+ * Uses {@link CsrfTokenRepository} to add the {@link CsrfLogoutHandler}.
+ *
+ * @author Josh Cummings
+ * @since 5.6
+ * @see Saml2LogoutConfigurer
+ */
+public final class Saml2LogoutConfigurer<H extends HttpSecurityBuilder<H>>
+		extends AbstractHttpConfigurer<Saml2LogoutConfigurer<H>, H> {
+
+	private ApplicationContext context;
+
+	private List<LogoutHandler> logoutHandlers = new ArrayList<>();
+
+	private SecurityContextLogoutHandler contextLogoutHandler = new SecurityContextLogoutHandler();
+
+	private String logoutSuccessUrl = "/login?logout";
```

Comment:
They are in separate packages, so I think that duplication is to be expected. Conceivably, `Saml2LogoutConfigurer` could be declared in the same package as `LogoutConfigurer`, though this departs from OAuth and existing SAML DSL support.

Regarding whether it's strange that logout does not impact saml2Logout, the way I see it is that these are separate logout mechanisms. Spring Security has numerous login mechanisms and it's not surprising that configuring one login mechanism does not affect another.

Reply:
One way to possibly overcoming this is making the `LogoutConfigurer` components, specifically `LogoutSuccessHandler` and the `LogoutHandler`s, public so that `Saml2LogoutConfigurer` can read them. The latest commit demonstrates this, and I think that it provides a good amount of simplification for a small API change.